File: pipeline/kite9-pipeline-common/src/main/java/com/kite9/k9server/adl/holder/meta/MetaWrite.java
package com.kite9.k9server.adl.holder.meta;

import java.net.URI;
import java.util.List;

import org.springframework.security.core.Authentication;

public interface MetaWrite {
	
	public static String CONTENT_CHANGED = "content-changed";

	public void setUser(UserMeta a);
	
	public void setUser();
	
	public void setAuthor(UserMeta a);
	
	public void setAuthorAndNotification(Authentication a);
	
	public void setTopicUri(URI topic);
	
	public void setCloseUri(URI close);
	
	public void setTitle(String title);
	
	public void setUri(URI u);
	
	public void setCollaborators(List<UserMeta> collaborators);
	
	public void setNotification(String message);
	
	public void setError(String message);
	
	public void setCommitCount(int c);
	
	public void setRole(Role r);
	
	public void setUploadsPath(String u);

	
}

File: pipeline/kite9-pipeline-common/src/main/kotlin/com/kite9/k9server/adl/holder/meta/MetaWrite.kt
package com.kite9.k9server.adl.holder.meta

import com.kite9.k9server.uri.URI

interface MetaWrite {

    fun setUser(a: UserMeta)
    fun setUser()
    fun setAuthor(a: UserMeta)
    fun setTopicUri(topic: URI)
    fun setCloseUri(close: URI)
    fun setTitle(title: String)
    fun setUri(u: URI)
    fun setCollaborators(collaborators: List<UserMeta>)
    fun setNotification(message: String)
    fun setError(message: String)
    fun setCommitCount(c: Int)
    fun setRole(r: Role)
    fun setUploadsPath(u: String)

    companion object {
        const val CONTENT_CHANGED = "content-changed"
    }
}